docs/sync-inventory-levels-to-variant-metafields/script.liquid

```diff
@@ -1,75 +1,205 @@
+          id
+          available
+          item {
+            variant {
+              id
+            }
+          }
+        }
+      }
+    {% endcapture %}
+
+    {% assign result = query | shopify %}
+
+    {% if event.preview %}
+      {% capture result_json %}
         {
-          "metafield": {
-            "namespace": {{ metafield_namespace | json }},
-            "key": {{ options.location_ids_and_metafield_keys__keyval_required.first.last | json }},
-            "value_type": "integer",
-            "value": 10
+          "data": {
+            "inventoryLevel": {
+              "available": 20,
+              "item": {
+                "variant": {
+                  "id": "gid://shopify/ProductVariant/1234567890"
+                }
+              }
+            }
           }
         }
-      ]
-    }
-  }
-{% elsif event.topic contains "shopify/inventory_levels/" %}
-  {% assign inventory_level_string = inventory_level.location_id | append: "" %}
+      {% endcapture %}
 
-  {% assign metafield_namespace = options.metafield_namespace__required | strip %}
-  {% assign metafield_key = options.location_ids_and_metafield_keys__keyval_required[inventory_level_string] %}
+      {% assign result = result_json | parse_json %}
+    {% endif %}
 
-  {% if metafield_key != blank %}
-    {
-      "action": {
-        "type": "shopify",
-        "options": [
-          "post",
-          "/admin/products/{{ inventory_level.variant.product_id }}/variants/{{ inventory_level.variant.id }}/metafields.json",
-          {
-            "metafield": {
-              "namespace": {{ metafield_namespace | json }},
-              "key": {{ metafield_key | json }},
-              "value_type": "integer",
-              "value": {{ inventory_level.available | json }}
+    {% action "shopify" %}
+      mutation {
+        metafieldsSet(
+          metafields: [
+            {
+              ownerId: {{ result.data.inventoryLevel.item.variant.id | json }}
+              namespace: {{ metafield_namespace | json }}
+              key: {{ metafield_key | json }}
+              type: "number_integer"
+              value: {{ result.data.inventoryLevel.available | append: "" | json }}
             }
+          ]
+        ) {
+          metafields {
+            id
+            namespace
+            key
+            type
+            value
+            owner {
+              ... on ProductVariant {
+                id
+              }
+            }
+          }
+          userErrors {
+            code
+            field
+            message
           }
-        ]
+        }
       }
-    }
+    {% endaction %}
   {% endif %}
+
 {% elsif event.topic == "mechanic/user/trigger" %}
-  {% for variant in shop.variants %}
-    {% for inventory_level in variant.inventory_levels %}
-      {% assign inventory_level_string = inventory_level.location_id | append: "" %}
+  {% capture bulk_operation_query %}
+    query {
+      productVariants(
+        query: ""
```

Comment:
@tekhaus Is there a purpose to this bit?

Reply:
Ah, yeah, sorry. I was considering adding a query filter option field, and ultimately decided not to, and accidentally left that line in.

Reply:
Removed in latest commit